### DISCUSSION TITLE
Authenticating user in Electron app via default browser.

### DISCUSSION THREAD

**Discussion by Jgrtowy** [supabase/supabase#22270] [Questions]
Hello,
I want to build an Electron (React) app with Supabase as my database and authentication service. 
However, when invoking the `signInWithOAuth` function it redirects user to login page inside of the embedded browser. 
Is there any way to redirect user to their default browser instead of the Electron app to authenticate then come back to Electron and let the function complete it's job?

**Comment by igorlanko** [comment]
More of an Electron question to be honest, but you can do it with:
```ts
import { shell } from 'electron'

shell.openExternal('https://your-site-url.com/auth')
```
https://www.electronjs.org/docs/latest/api/shell

Then your website deep links back to your app's URL schema `your.app.schema://auth?someAuthParam=value`. Schema is required so that the browser sends a signal to the system to open your installed app. Registering a schema depends on your packager, even though it's relatively straightforward in most. For example `appId` in `electron-builder` https://www.electron.build/configuration/configuration.

And you can listen to this redirect with:
```ts
app.on('second-instance', () => {
	const url = process.argv.find((arg) => arg.startsWith('your.app.schema://'))
	if (url) handleAuthUrl(url)
})

app.on('open-url', (event, url) => {
	event.preventDefault()
	handleAuthUrl(url)
})
```
in the main process. `handleAuthUrl()` apparently could deconstruct the session from the URL. 

However in the context of Supabase, I'm not entirely sure what exactly that deep link URL entails, that's the question I currently don't have the answer for https://github.com/orgs/supabase/discussions/27181.
